### PR TITLE
[bitnami/thanos] adding ability to pass env variables to containers

### PR DESCRIPTION
**Pull request by doug-ba**
**Description of the change**

Allows optional passing of environment variables to containers.
**Benefits**
For my use case this will allow passing `HTTPS_PROXY`, `HTTP_PROXY` and `NO_PROXY` environment variables to allow thanos containers inside our private cloud to connect to google APIs via our squid proxy.

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/thanos/Chart.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ name: thanos
 sources:
   - https://github.com/bitnami/bitnami-docker-thanos
   - https://thanos.io
-version: 3.14.1
+version: 3.14.2
```
Comment (migruiz4) [inline bitnami/thanos/Chart.yaml:31]: As this PR adds multiple settings, could you please replace the version with a minor version bump `3.15.0`?

File: bitnami/thanos/templates/ruler/statefulset.yaml [hunk 2]
```diff
@@ -109,10 +109,13 @@ spec:
...
             {{- end }}
           env:
-            - name: POD_NAME
-              valueFrom:
-                fieldRef:
-                  fieldPath: metadata.name
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
```
Comment (migruiz4) [inline bitnami/thanos/templates/ruler/statefulset.yaml:115]: Please revert the changes to the POD_NAME env variable and use `nindent 12` instead

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @doug-ba ,

Thank you very much for your contribution.

Your PR looks good to me, but there is an issue with the indentation.
Using `nindent 10` will have the following result:
```
          env:
          - name: VARNAME1
            value: value1
          - name: VARNAME2
            valueFrom:
              secretKeyRef:
                key: varname2-key
                name: existing-secret
```
Instead, could you please use `nindent 12` so the yaml is indented like this?
```
          env:
            - name: VARNAME1
              value: value1
            - name: VARNAME2
              valueFrom:
                secretKeyRef:
                  key: varname2-key
                name: existing-secret
```

If it is okay for you, I will add a commit fixing this for `thanos.receive`:
```
          env:
          - name: NAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
          - name: NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace
          - name: OBJSTORE_CONFIG
            valueFrom:
              secretKeyRef:
                key: objstore.yml
                name: RELEASE-NAME-thanos-objstore-secret
          - name: VARNAME1
            value: value1
          - name: VARNAME2
            valueFrom:
              secretKeyRef:
                key: varname2-key
                name: existing-secret
```

**Comment by migruiz4** [comment]
Additionally, in case you haven't noticed, your PR is changing the permissions of all the modified files from 100644 → 100755

File: bitnami/thanos/Chart.yaml (hunk 1 above)
Comment (doug-ba) [reply bitnami/thanos/Chart.yaml]: Bumped to 3.15.0

File: bitnami/thanos/templates/ruler/statefulset.yaml (hunk 2 above)
Comment (doug-ba) [reply bitnami/thanos/templates/ruler/statefulset.yaml]: I reverted the indentation change, but either works.

**Review by marcosbc — APPROVED** [review]
LGTM, thanks for the contribution!

**Review by marcosbc — CHANGES_REQUESTED** [review]
Actually, could you fix the merge conflict? If not we're unable to merge this.

**Comment by doug-ba** [comment]
Corrected the merge conflict in Chart.yaml.  Hopefully it doesn't change again before you folks get to merging.

File: bitnami/thanos/templates/bucketweb/deployment.yaml [hunk 3]
```diff
@@ -76,6 +76,10 @@ spec:
             {{- if .Values.bucketweb.extraFlags }}
             {{- .Values.bucketweb.extraFlags | toYaml | nindent 12 }}
             {{- end }}
+          {{- if .Values.bucketweb.extraEnv }}
+          env:
+            {{- toYaml .Values.bucketweb.extraEnv | nindent 12 }}
```
Comment (marcosbc) [inline bitnami/thanos/templates/bucketweb/deployment.yaml:81]: Something I just noticed is that we're using `nindent` here. The difference between `nindent` and `indent` is that it begins with a newline, so the rendered output would look like this:

```yaml
env:
  
  - name: env1
    value: test
```

Could you fix this and other lines where `nindent` is used? I.e.:

```suggestion
          env: {{- toYaml .Values.bucketweb.extraEnv | nindent 12 }}
```

**Review by marcosbc — CHANGES_REQUESTED** [review]
Another request, just noticed this.

**Review by marcosbc — APPROVED** [review]
Actually nevermind, I see that the space removal (from "`{{-`") would actually make it render properly. So LGTM now! Thanks a lot!
